### DISCUSSION TITLE
Improving error messages

### DISCUSSION THREAD

**Discussion by salihgueler** [GetStream/stream-feed-flutter#107] [Ideas]
This is an idea/Q&A topic actually. I appreciate that you are directing users to the links for errors but, this can be difficult to follow sometimes.

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: StreamApiException{body: {detail: You do not have permission to do this, you got this error because there are no policies allowing this request on this application. Please consult the documentation https://getstream.io/docs/, status_code: 403, code: 17, exception: NotAllowedException, duration: 0.19ms, more_info: https://getstream.io/docs/api_error_responses}, jsonData: null, status: 403, code: null}
```

This states that I do not have permission to do this and there are no policies allowing the request. As a user then I would do the following:

- Go to the links
- API Error Responses page shows exactly the same information without any solution (even for common scenarios)
- Generic document page is way broader to find a solution.

Then I would ask the following questions:

- What are policies?
- Why am I not able to do the operation since I am already on the feeds program and I did not pass my quota?
- How can I fix this?

I believe the messages should **at least** either take the user to exact page about information/solution or tell possible solutions like "Be sure you are part of the project." etc.

------

BTW, I could not fix the problem above as well. So if you can help me out, that would be great as well :)

**Comment by sachaarbonel** [comment]
Hey @salihgueler, as I understand you would like more helpful error messages right? For now, we just pass down the message from the backend but we could send more helpful links based on [stream code and status code](https://getstream.io/activity-feeds/docs/node/api_error_responses/?q=error%20code). Your error looks like you are trying things that are not permitted by the default permission policies in [place](https://getstream.io/activity-feeds/docs/node/auth_and_permissions/). It is here to prevent misuse of the SDK, but if you'd like custom policies for your use case support will be able to help you with this? Btw what queries did you try to run?

**Reply by salihgueler** [reply]
Did you change something in the security policies though? Our backend generates a token and we use the same token for both Chat and Feed. At least I am sure I used it for both while doing proof of concept. Do we need to generate two tokens for two products?

I simply want to add an activity to a notification feed. Nothin complicated, to be honest.

**Reply by salihgueler** [reply]
Also, when I try to do the operations with the token generated by the token generator (https://getstream.io/chat/docs/react/token_generator/), I am having the same problem as well. This way I could eliminate a possible problem on backend.
